src/lib/plugins.c

```diff
@@ -40,6 +40,21 @@ void bd_plugin_spec_free (BDPluginSpec *spec) {
     g_free (spec);
 }
 
+/**
+ * bd_plugin_spec_new: (constructor)
+ * @name: %BDPlugin name, e.g. %BD_PLUGIN_LVM
+ * @so_name: (nullable): SO name of the plugin to load or %NULL for default
+ *
+ * Returns: (transfer full): a new plugin spec
+ */
+BDPluginSpec* bd_plugin_spec_new (BDPlugin name, const gchar *so_name) {
+    BDPluginSpec* ret = g_new0 (BDPluginSpec, 1);
+    ret->name = name;
+    ret->so_name = so_name ? g_strdup (so_name) : NULL;
```

Comment:
allocating a const? typecasting it for `g_free()`?

Reply:
I know it's weird, but we already allocate it in `bd_plugin_spec_copy` and free it in `bd_plugin_spec_free` and I didn't want to change it.

Reply:
Hmm fair enough. Perhaps you could add some `/* FIXME */`s? :-)